lntest/itest/lnd_wallet_import_test.go

```diff
@@ -0,0 +1,784 @@
+	}{
+		{
+			name:     "standard BIP-0049",
+			addrType: walletrpc.AddressType_NESTED_WITNESS_PUBKEY_HASH,
+		},
+		{
+			name:     "lnd BIP-0049 variant",
+			addrType: walletrpc.AddressType_HYBRID_NESTED_WITNESS_PUBKEY_HASH,
+		},
+		{
+			name:     "standard BIP-0084",
+			addrType: walletrpc.AddressType_WITNESS_PUBKEY_HASH,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		success := t.t.Run(tc.name, func(tt *testing.T) {
+			ht := newHarnessTest(tt, net)
+			ht.RunTestCase(&testCase{
+				name: tc.name,
+				test: func(net1 *lntest.NetworkHarness, t1 *harnessTest) {
+					testWalletImportAccountScenario(
+						net, t, tc.addrType,
+					)
+				},
+			})
+		})
+		if !success {
+			// Log failure time to help relate the lnd logs to the
+			// failure.
+			t.Logf("Failure time: %v", time.Now().Format(
+				"2006-01-02 15:04:05.000",
+			))
+			break
+		}
+	}
+}
+
+func testWalletImportAccountScenario(net *lntest.NetworkHarness, t *harnessTest,
+	addrType walletrpc.AddressType) {
+
+	ctxb := context.Background()
+	const utxoAmt int64 = btcutil.SatoshiPerBitcoin
+
+	// We'll start our test by having two nodes, Carol and Dave. Carol's
+	// default wallet account will be imported into Dave's node.
+	carol, err := net.NewNode("carol", nil)
+	require.NoError(t.t, err)
+	defer shutdownAndAssert(net, t, carol)
+
+	dave, err := net.NewNode("dave", nil)
+	require.NoError(t.t, err)
+	defer shutdownAndAssert(net, t, dave)
+
+	ctxt, cancel := context.WithTimeout(ctxb, defaultTimeout)
+	defer cancel()
+	listReq := &walletrpc.ListAccountsRequest{
+		Name:        "default",
+		AddressType: addrType,
+	}
+	listResp, err := carol.WalletKitClient.ListAccounts(ctxt, listReq)
+	require.NoError(t.t, err)
+	require.Equal(t.t, len(listResp.Accounts), 1)
+	carolAccount := listResp.Accounts[0]
+
+	ctxt, cancel = context.WithTimeout(ctxb, defaultTimeout)
+	defer cancel()
+	const importedAccount = "carol"
+	importReq := &walletrpc.ImportAccountRequest{
+		Name:              importedAccount,
+		ExtendedPublicKey: carolAccount.ExtendedPublicKey,
+		AddressType:       addrType,
+	}
+	_, err = dave.WalletKitClient.ImportAccount(ctxt, importReq)
+	require.NoError(t.t, err)
+
+	// We'll generate an address for Carol from Dave's node to receive some
+	// funds.
+	ctxt, cancel = context.WithTimeout(ctxb, defaultTimeout)
+	defer cancel()
+	daveAddrResp, err := dave.NewAddress(ctxt, &lnrpc.NewAddressRequest{
+		Type:    walletToLNAddrType(t.t, addrType),
+		Account: importedAccount,
+	})
+	require.NoError(t.t, err)
+
+	// We'll also generate the same address for Carol, as it'll be required
+	// later when signing.
+	ctxt, cancel = context.WithTimeout(ctxb, defaultTimeout)
+	defer cancel()
+	carolAddrResp, err := carol.NewAddress(ctxt, &lnrpc.NewAddressRequest{
+		Type: walletToLNAddrType(t.t, addrType),
+	})
+	require.NoError(t.t, err)
+
+	// Sanity check that the generated addresses match.
+	require.Equal(t.t, daveAddrResp.Address, carolAddrResp.Address)
```

Comment:
Due to the 3 different constellations we have now, should we add an address type validation here and for the change output below? I think it should be quite easy to add with `btcutil.DecodeAddress` but would've caught the incorrect type for the hybrid one.